conda/core/solve.py

```diff
@@ -420,7 +421,8 @@ def _add_specs(self, ssc):
                 elif ssc.update_modifier == UpdateModifier.FREEZE_INSTALLED:
                     if pkg_name in context.aggressive_update_packages:
                         ssc.specs_map[pkg_name] = pkg_name
-                    elif target_prec in explicit_spec_package_pool:
+                    elif (target_prec in explicit_spec_package_pool or
+                          target_prec.name not in explicit_spec_package_pool_names):
```

Comment:
What's the difference between `explicit_spec_package_pool` and `explicit_spec_package_pool_names`? From the naming it seems like they should represent pretty much the same data?

Reply:
These are two different set lookups.  The first is an exact match.  If it finds this match, then freezing things is definitely OK.  The second is an "is this package related to anything I'm doing?" check.  If it's not related, we can also freeze this package, because it shouldn't be affected by what we're trying to do.

Reply:
ahh, gotcha!